### PR TITLE
Uninitialized attributes will ERROR instead of returning as Nil

### PR DESCRIPTION
**Pull request by bfrey08**
Previous implementation:

```
WhizBang.select(:id).first.name => nil
```

After this PR:

```
WhizBang.select(:id).first.name => ActiveModel::MissingAttributeError "missing attribute: name"
```

The new implementation is more like how Rails handles uninitialized attributes. It is significantly safer, as when we are developing with .select statements we are more likely to error out and not unknowingly need an attribute that we are calling nil.

### CODE REVIEW

File: lib/active_force/sobject.rb [hunk 1]
```diff
@@ -62,6 +63,9 @@ def self.describe
     def self.build mash, association_mapping={}
       return unless mash
       sobject = new
+
+      not_selected_attributes = sobject.class.fields.reject{|key| mash.keys.include?(key)}
```
Comment (asedge) [inline lib/active_force/sobject.rb:67]: Do you think `attributes_not_selected` is more readable than `not_selected_attributes`?

File: lib/active_model/uninitialized_sobject.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require 'active_model'
+
+module ActiveModel
+  class Attribute
+    class UninitializedSobject < Uninitialized # :nodoc:
```
Comment (asedge) [inline lib/active_model/uninitialized_sobject.rb:5]: Not loving the name of this class. Maybe `UninitializedValue` makes more sense? I dunno.

File: lib/active_force/sobject.rb [hunk 3]
```diff
@@ -108,6 +112,20 @@ def create!
...
+          if attrs.include?(self.mappings.dig(key.name.to_sym))
+            ActiveModel::Attribute::UninitializedSobject.new(key.name, key.type)
+          else
+            key
+          end
+        end
+      attrs = attr_set.instance_variable_get(:@attributes) || {}
+      new_attr_set = ActiveModel::LazyAttributeSet.new({}, {}, {}, {}, attrs)
+      self.instance_variable_set(:@attributes, new_attr_set)
+    end
```
Comment (asedge) [inline lib/active_force/sobject.rb:127]: Not loving this dance of replacing the all the attributes with the `LazyAttributeSet` class.  Just curious... do you actually need that or can you use the simpler `ActiveModel::AttributeSet` instance that we already have?  Looking through the code there doesn't seem to be much special about the `ActiveModel::LazyAttributeSet` class for this specific use case.  Could you do something like this?

```
def set_uninitialized_attributes(attrs)
  return if attrs.blank?
  attrs.each do |attr|
    attributes[attr] = ActiveModel::Attribute::UninitializedSobject.new(attributes[attr].name, attributes[attr].type)
  end
end
```

File: lib/active_force/sobject.rb (hunk 1 above)
Comment (bfrey08) [reply lib/active_force/sobject.rb:67]: I like it

File: lib/active_model/uninitialized_sobject.rb (hunk 2 above)
Comment (bfrey08) [reply lib/active_model/uninitialized_sobject.rb:5]: Good call and I agree. It's more like `UninitializedSobjectAttribute`; but that is so lengthy. I'll change it to `UninitializedValue`

File: lib/active_force/sobject.rb (hunk 3 above)
Comment (asedge) [reply lib/active_force/sobject.rb:127]: Happy to pair on this... I _think_ I had something working in the REPL.

Comment (bfrey08) [reply lib/active_force/sobject.rb]: Good call. I was able to mutate the attributes by accessing the @attributes instance variable inside of the @attributes instance variable. See code changes

File: lib/active_force/sobject.rb [hunk 4]
```diff
@@ -7,7 +7,7 @@
 require 'forwardable'
 require 'logger'
 require 'restforce'
-require 'active_model/uninitialized_sobject.rb'
+require 'active_model/uninitialized_value.rb'
```
Comment (asedge) [inline lib/active_force/sobject.rb:10]: The `.rb` shouldn't be necessary here and it's not common that it's included.  Would you mind chopping it off, please?

**Comment by asedge** [comment]
@bfrey08 This is shaping up nicely! Any chance we could get some specs to verify this new feature?

File: spec/active_force/sobject/select_spec.rb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        response = [build_restforce_sobject({
+          "Id"       => "123",
+          "Quota__c" => "321",
+        })]
+        allow(client).to receive(:query).once.and_return response
+        territory = Territory.select(:id).first
+        expect(territory.instance_variable_get(:@attributes)["name"]).to be_an_instance_of(ActiveModel::Attribute::UninitializedValue)
+      end
+
+      it 'raises missing attribute error if uninitialized variable is called' do
```
Comment (asedge) [inline spec/active_force/sobject/select_spec.rb:27]: This should be a context.  Something like:
```
context 'when getting the value for an uninitialized variable' do
  it 'raises an error' do
```

**Review by asedge — APPROVED** [review]
LGTM, nice work dude!

**Pull request merged**